### PR TITLE
Change buffer in wolfSSL_dtls_import to be const

### PR DESCRIPTION
**Pull request by julek-wolfssl**

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -1346,7 +1346,7 @@ int wolfSSL_dtls_export_internal(WOLFSSL* ssl, byte* buf, word32 sz)
 
 
 /* On success return amount of buffer consumed */
-int wolfSSL_dtls_import_state_internal(WOLFSSL* ssl, byte* buf, word32 sz)
+int wolfSSL_dtls_import_state_internal(WOLFSSL* ssl, const byte* buf, word32 sz)
```
Comment (ejohnstown) [inline src/internal.c:1349]: scan-build is reporting the value of idx  from the line `idx += ret;` at the end of the function isn't being used. It is returning `ret` and it should probably be returning `idx`.

**Review by ejohnstown — CHANGES_REQUESTED** [review]

**Review by ejohnstown — APPROVED** [review]

File: wolfssl/ssl.h [hunk 2]
```diff
@@ -743,7 +743,7 @@ typedef int (*wc_dtls_export)(WOLFSSL* ssl,
 #define WOLFSSL_DTLS_EXPORT_TYPES
 #endif /* WOLFSSL_DTLS_EXPORT_TYPES */
 
-WOLFSSL_API int wolfSSL_dtls_import(WOLFSSL* ssl, unsigned char* buf,
+WOLFSSL_API int wolfSSL_dtls_import(WOLFSSL* ssl, const unsigned char* buf,
```
Comment (toddouska) [inline wolfssl/ssl.h:746]: What's driving this change to a const input parameter?

**Comment by julek-wolfssl** [comment]
A customer asked for this change to be able to pass in a pointer to Flash storage to ensure no writes to that memory.
-------- Original message --------From: toddouska <notifications@github.com> Date: 29/09/2020  22:40  (GMT+01:00) To: wolfSSL/wolfssl <wolfssl@noreply.github.com> Cc: julek-wolfssl <juliusz@wolfssl.com>, State change <state_change@noreply.github.com> Subject: Re: [wolfSSL/wolfssl] Change buffer in wolfSSL_dtls_import to be const (#3221) 
@toddouska commented on this pull request.



In wolfssl/ssl.h:
> @@ -743,7 +743,7 @@ typedef int (*wc_dtls_export)(WOLFSSL* ssl,
 #define WOLFSSL_DTLS_EXPORT_TYPES
 #endif /* WOLFSSL_DTLS_EXPORT_TYPES */
 
-WOLFSSL_API int wolfSSL_dtls_import(WOLFSSL* ssl, unsigned char* buf,
+WOLFSSL_API int wolfSSL_dtls_import(WOLFSSL* ssl, const unsigned char* buf,

What's driving this change to a const input parameter?

File: wolfssl/ssl.h (hunk 2 above)
Comment (julek-wolfssl) [reply wolfssl/ssl.h:746]: A customer asked for this change to be able to pass in a pointer to Flash storage to ensure no writes to that memory.

**Review by toddouska — APPROVED** [review]

**Pull request merged**